src/compiler/checker.ts

```diff
@@ -25771,14 +25771,21 @@ namespace ts {
                     error(node, Diagnostics.Value_of_type_0_is_not_callable_Did_you_mean_to_include_new, typeToString(funcType));
                 }
                 else {
-                    let relatedInformation: DiagnosticRelatedInformation | undefined;
-                    if (node.arguments.length === 1) {
+                    const relatedInformation: DiagnosticRelatedInformation[] = [];
+                    if (node.arguments.length === 0) {
+                        // Diagnose get accessors incorrectly called as functions
+                        const { resolvedSymbol } = getNodeLinks(node.expression);
+                        if (resolvedSymbol && resolvedSymbol.flags & SymbolFlags.GetAccessor) {
+                            relatedInformation.push(createDiagnosticForNode(node.expression, Diagnostics._0_is_a_get_accessor_did_you_mean_to_use_it_without, getTextOfNode(node.expression)));

```

Comment:
If you change the error message, you might want to add a diagnostic that points back to the declaration - we already have `Diagnostics._0_is_declared_here`.

Reply:
I wasn't sure how to get back to the original declaration. Would it be ok to add a special-case check here whether the `node.expression` is a PropertyAccessExpression? (seems a bit roundabout.)